### PR TITLE
Shuffle around the wiggle crates

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit reorganizes the wiggle crates slightly by performing the
following transforms:

* The `crates/wiggle` crate, previously named `wiggle`, was moved to
  `crates/wiggle/crates/macro` and is renamed to `wiggle-macro`.

* The `crates/wiggle/crates/runtime` crate, previously named
  `wiggle-runtime`, was moved to `crates/wiggle` and is renamed to
  `wiggle`.

* The new `wiggle` crate depends on `wiggle-macro` and reexports the macro.

The goal here is that consumers only deal with the `wiggle` crate
itself. No more crates depend on `wiggle-runtime` and all dependencies
are entirely on just the `wiggle` crate.

### CODE REVIEW

**Comment by pchickey** [comment]
Thanks! Do we want to keep the crates subdir inside wiggle any longer? We don't have that indirection in wasi-common. This may be a good opportunity to get rid of it.

File: scripts/test-all.sh [hunk 1]
```diff
@@ -62,7 +62,6 @@ RUST_BACKTRACE=1 cargo test \
   --package wasmtime-obj \
   --package wiggle \
   --package wiggle-generate \
-  --package wiggle-runtime \
   --package wiggle-test \
```
Comment (pchickey) [inline scripts/test-all.sh]: do you have to add wiggle-macro back here?

Comment (alexcrichton) [reply scripts/test-all.sh]: Sure yeah. TBH I [don't know the fate of this script](https://github.com/bytecodealliance/wasmtime/pull/1397) and it already tests very little compared to CI, but can't hurt to keep touching it.

Comment (pchickey) [reply scripts/test-all.sh]: thanks, i hadn't paid attention to all of that discussion.

**Comment by kubkon** [comment]
FWIW, it looks like the paths to the test `witx` files need to updated for the tests to pass.

**Pull request merged**